File: ObjcExample/TestProj/style/NestedSpinnerViewController.m
//
//  NestedSpinnerViewController.m
//  TestProj
//
//  Created by Nick on 22/09/21.
//  Copyright © 2021 Nick Wang. All rights reserved.
//

#import "NestedSpinnerViewController.h"

@interface NestedSpinnerViewController ()

@property (strong, nonatomic) NestedSpinnerView *spinner;

@end

@implementation NestedSpinnerViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = @"Nested Spinner";
    [self setup];
}

- (void)setup {
    [NestedSpinnerAppearance darkNested];
    _spinner = [[NestedSpinnerView alloc] initWithFrame:CGRectMake(50, 160, self.view.bounds.size.width - 100, 50)];
    _spinner.style.cellSelectedBackgroundColor = [UIColor colorWithWhite:0.3 alpha:0.2];
    _spinner.anchorTextFont = [UIFont boldSystemFontOfSize:18];
    _spinner.anchorTintColor = UIColor.darkTextColor;
    _spinner.cellNib = [UINib nibWithNibName:NSStringFromClass([NestedSubItemCell class]) bundle:NestedSpinnerBundle.bundle];
    NestedSpinnerItem *groupItem1 = [NestedSpinnerItem new];
    groupItem1.groupItemTitle = @"Basic Usage";
    groupItem1.subItems = @[ [[Entity alloc] initWithName:@"Normal Spinner" withData:@(1)], [[Entity alloc] initWithName:@"Nested Spinner" withData:@(2)], [[Entity alloc] initWithName:@"Programmatically" withData:@(3)] ];
    NestedSpinnerItem *groupItem2 = [NestedSpinnerItem new];
    groupItem2.groupItemTitle = @"Customised";
    groupItem2.subItems = @[ [[Entity alloc] initWithName:@"Custom Attributes" withData:@(4)], [[Entity alloc] initWithName:@"Delegate Colours" withData:@(5)], [[Entity alloc] initWithName:@"Custom AnchorView" withData:@(6)] ];
    _spinner.dataTrees = @[ groupItem1, groupItem2 ];
    _spinner.anchorOffset = CGPointMake(0, _spinner.bounds.size.height);
    _spinner.selectionAction = ^(NSIndexPath * _Nonnull indexPath, NSString * _Nonnull value, id _Nullable userdata) {
        NSLog(@"indexPath: %@, selected value:%@", indexPath, value);
        if ([userdata isKindOfClass:[Entity class]]) {
            Entity *entity = (Entity *)userdata;
            
            switch (entity.data.intValue) {
                case 1:
                    break;
                
                default:
                    break;
            }
            
        }
    };
    [self.view addSubview:_spinner];
}

@end

File: Example/NestedSpinner/Style/NestedSpinnerViewController.swift
//
//  NestedSpinnerViewController.swift
//  NestedSpinner
//
//  Created by Nick on 22/09/21.
//  Copyright © 2021 Nick Wang. All rights reserved.
//

import UIKit
import NestedSpinner

class NestedSpinnerViewController: UIViewController {
    
    var spinner: NestedSpinnerView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = "Nested Spinner"
        setup()
    }
    
    func setup() {
        NestedSpinnerAppearance.darkNested()
        spinner = NestedSpinnerView(frame: CGRect(x: 50, y: 160, width: self.view.bounds.width - 100, height: 50))
        spinner.style.cellSelectedBackgroundColor = UIColor(white: 0.3, alpha: 0.2)
        spinner.anchorTextFont = UIFont.boldSystemFont(ofSize: 18)
        spinner.anchorTintColor = .darkText
        spinner.cellNib = UINib(nibName: String(describing: NestedSubItemCell.self), bundle: NestedSpinnerBundle.bundle)
        let groupItem1 = NestedSpinnerItem("Basic Usage", [Entity("Normal Spinner", 1), Entity("Nested Spinner", 2), Entity("Programmatically", 3)])//"Init Expanded, Colours"
        let groupItem2 = NestedSpinnerItem("Customised", [Entity("Custom Attributes", 4), Entity("Delegate Colours", 5), Entity("Custom AnchorView", 6)])
        spinner.dataTrees = [groupItem1, groupItem2]
        spinner.anchorOffset = CGPoint(x: 0, y: spinner.bounds.size.height)
        spinner.selectionAction = { indexPath, value, userdata in
            print("indexPath: \(indexPath), selected value:\(value)")
            if userdata is Entity {
                let entity = userdata as! Entity
                switch entity.data {
                case 1:
                    break
                default:
                    break
                }
                
            }
        }
        self.view.addSubview(spinner)
    }
    
}
